### PR TITLE
Add linting to dashboard

### PR DESCRIPTION
**Pull request by hulloanson**
Added linting to dashboard by adapting GTS.

Also edited existing code to fix linting errors afterwards

### CODE REVIEW

File: bikespace_dashboard/js/app.js [hunk 1]
```diff
@@ -1,20 +1,19 @@
...
-      let summary_box = new SummaryBox("#sidebar", "summary-box", shared_state);
-      let issue_chart = new IssueChart("#sidebar", "issue-chart", shared_state);
-      let day_chart = new DayChart("#sidebar", "day-chart", shared_state);
-      let duration_tod_chart = new DurationTimeOfDayChart("#sidebar", "duration-tod-chart", shared_state);
-      let map = new Map("body", "map", shared_state);
-  }
-});
\ No newline at end of file
+  success: function (data) {
+    const shared_state = new SharedState(data);
```
Comment (tallcoleman) [inline bikespace_dashboard/js/app.js:11]: Need to double check assigning `shared_state` as const doesn't cause any unintended behaviour

File: bikespace_dashboard/js/components/map.js [hunk 2]
```diff
@@ -1,140 +1,141 @@
...
-      'overnight': "<strong>overnight</strong>",
-      'multiday': "for <strong>several days</strong>"
+      minutes: 'for <strong>less than an hour</strong>',
+      hours: 'for <strong>several hours</strong>',
+      overnight: '<strong>overnight</strong>',
+      multiday: 'for <strong>several days</strong>',
     };
-    let issue_tags = {};
-    for (let entry of Object.values(ia)) {
+    const issue_tags = {};
```
Comment (tallcoleman) [inline bikespace_dashboard/js/components/map.js:74]: Same as earlier, `issue_tags` has values assigned to it later, so const feels wrong, but I guess it works?

File: bikespace_dashboard/js/components/map.js [hunk 3]
```diff
@@ -1,140 +1,141 @@
...
+        `<p class="submission-id">ID: ${point.id}</p>`,
+      ].join('');
 
       // BUILD MARKERS
       // set up custom markers
-      let BaseIcon = L.Icon.extend({
+      const BaseIcon = L.Icon.extend({
         options: {
-          shadowUrl: "./leaflet/images/marker-shadow.png",
-          iconSize:     [36, 36],
```
Comment (tallcoleman) [inline bikespace_dashboard/js/components/map.js:118]: I actually prefer the aligned brackets here

Comment (hulloanson) [reply bikespace_dashboard/js/components/map.js:118]: sorry, must have removed the line during a merge.

Comment (hulloanson) [reply bikespace_dashboard/js/components/map.js:118]: let's see what I can do with prettier...

File: bikespace_dashboard/js/components/map.js (hunk 2 above)
Comment (hulloanson) [reply bikespace_dashboard/js/components/map.js:74]: `issue_tag`'s value holds the reference to an object. Assigning values to the object's properties do not change its value so a `const` was used

File: bikespace_dashboard/js/app.js (hunk 1 above)
Comment (hulloanson) [reply bikespace_dashboard/js/app.js:11]: `shared_state` holds a constant reference to a SharedState instance and is never reassigned. so we can safely use `const` here

File: bikespace_dashboard/js/components/map.js (hunk 3 above)
Comment (hulloanson) [reply bikespace_dashboard/js/components/map.js:118]: prettier's devs explicitly stated they [would not support vertical alignment](https://github.com/prettier/prettier/issues/188#issuecomment-272627905) and it'd be too much effort to use either eslint's [vertical alignment plugin](https://www.npmjs.com/package/eslint-plugin-align-assignments) along with prettier.

would therefore leave it at this

**Comment by hulloanson** [comment]
@tallcoleman @sonalranjit this PR should be ready then. see if you could review it again, test it out and merge

**Review by sonalranjit — APPROVED** [review]
Looks good to me @hulloanson !
FYI: github actions don't work on forked PRs because it would expose the env variables for cloudflare and fly

**Pull request merged**